src/main/kotlin/frc/robot/subsystems/intake/extender/Extender.kt

```diff
@@ -0,0 +1,62 @@
+package frc.robot.subsystems.intake.extender
+
+import edu.wpi.first.units.Units
+import edu.wpi.first.units.measure.Distance
+import edu.wpi.first.wpilibj2.command.Command
+import edu.wpi.first.wpilibj2.command.SubsystemBase
+import edu.wpi.first.wpilibj2.command.button.Trigger
+import org.littletonrobotics.junction.AutoLogOutput
+import org.littletonrobotics.junction.Logger
+
+class Extender(private val io: ExtenderIO) : SubsystemBase() {
+
+    private fun setPosition(position: Distance) = runOnce {
+        io.setPosition(position)
+    }
+
+    private fun setPower(power: Double) = runOnce {
+        io.setPower(power)
+    }
+
+    fun extend() = setPosition(EXTENDED_POSITION)
+
+    fun retract() = setPosition(RETRACTED_POSITION)
+
+    fun reset(): Command {
+        return setPower(RESET_POWER)
+            .until(isStuck)
+            .andThen(
+                setPower(0.0),
+                runOnce { io.reset() }
```

Comment:
```suggestion
                runOnce(io::reset)
```
I'm on my phone but it think it works

Reply:
05e7f736641bd4ea24209d20878ed6bd7b8d3556